src/posit/connect/groups.py

```diff
@@ -0,0 +1,201 @@
+"""Group resources."""
+
+from __future__ import annotations
+from typing import List, overload
+
+import requests
+
+from . import me, urls
+
+from .config import Config
+from .paginator import Paginator
+from .resources import Resource, Resources
+
+
+class Group(Resource):
+    """Group resource.
+
+    Attributes
+    ----------
+    guid : str
+    name: str
+    owner_guid: str
+    """
+
+    @property
+    def guid(self) -> str:
+        return self.get("guid")  # type: ignore
+
+    @property
+    def name(self) -> str:
+        return self.get("name")  # type: ignore
+
+    @property
+    def owner_guid(self) -> str:
+        return self.get("owner_guid")  # type: ignore
+
+    # CRUD Methods
+
+    def delete(self) -> None:
+        """Delete the group."""
+        path = f"v1/groups/{self.guid}"
+        url = urls.append(self.config.url, path)
+        self.session.delete(url)
+
+
+class Groups(Resources):
+    """Groups resource."""
+
+    def __init__(self, config: Config, session: requests.Session) -> None:
+        self.config = config
+        self.session = session
+
+    @overload
+    def create(self, name: str, unique_id: str | None) -> Group:
+        """Create a group.
+
+        Parameters
+        ----------
+        name: str
+        unique_id: str | None
+
+        Returns
+        -------
+        Group
+        """
+        ...
+
+    @overload
+    def create(self, *args, **kwargs) -> Group:
+        """Create a group.
+
+        Returns
+        -------
+        Group
+        """
+        ...
+
+    def create(self, *args, **kwargs) -> Group:
+        """Create a group.
+
+        Parameters
+        ----------
+        name: str
+        unique_id: str | None
+
+        Returns
+        -------
+        Group
+        """
+        ...
+        body = dict(*args, **kwargs)
+        path = "v1/groups"
+        url = urls.append(self.config.url, path)
```

Comment:
Any reason to have `path` as a separate variable?

Reply:
not here, but comes into play downstream: https://github.com/posit-dev/posit-sdk-py/pull/228/files#diff-b7d2a2eb92d17a5b4aff237de4af47283494386561a1e96fea70f0d5ba4201d8R207-R211